### PR TITLE
Build golang container for both Ubuntu 18.04 and 20.04

### PR DESCRIPTION
**Pull request by ymmt2005**
and drop golang114.  It also stop vendoring dependencies.

### CODE REVIEW

File: golang/Dockerfile [hunk 1]
```diff
@@ -24,7 +24,7 @@ RUN curl -s -f -O https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz \
 
 COPY ./analyzer /work/analyzer
 WORKDIR /work/analyzer
-RUN GOBIN=/usr/local/bin go install -mod=vendor ./cmd/...
+RUN GOBIN=/usr/local/bin go install -mod=mod ./cmd/...
```
Comment (ymmt2005) [inline golang/Dockerfile:27]: I couldn't figure out why the builds failed w/o `-mod=mod`.
Only when I add this, the build passes...

Comment (ymmt2005) [reply golang/Dockerfile:27]: Wait, what's this.
```console
$ ls -al /usr/local/go115/src/vendor/golang.org/x/
total 24
drwxr-xr-x 6 root root 4096 Oct 14 19:16 .
drwxr-xr-x 3 root root 4096 Oct 14 19:16 ..
drwxr-xr-x 9 root root 4096 Oct 14 19:16 crypto
drwxr-xr-x 9 root root 4096 Oct 14 19:16 net
drwxr-xr-x 3 root root 4096 Oct 14 19:16 sys
drwxr-xr-x 5 root root 4096 Oct 14 19:16 text
```

Comment (ymmt2005) [reply golang/Dockerfile]: It was due to a bug in CircleCI built-in git client.
Addressed by installing git with apk.

**Review by morimoto-cybozu — APPROVED** [review]
LGTM

**Pull request merged**